hbase-server/src/main/java/org/apache/hadoop/hbase/master/replication/ReplicationPeerManager.java

```diff
@@ -501,6 +511,32 @@ private void checkClusterKey(String clusterKey) throws DoNotRetryIOException {
     }
   }
 
+  private void checkClusterId(String clusterKey) throws DoNotRetryIOException {
+    String peerClusterId = "";
+    try {
+      // Create the peer cluster config for get peer cluster id
+      Configuration peerConf = HBaseConfiguration.createClusterConf(conf, clusterKey);
+      ZKWatcher zkWatcher =
+        new ZKWatcher(peerConf, this + "check-peer-cluster-key", new Abortable() {
+          @Override public void abort(String why, Throwable e) {
+          }
+
+          @Override public boolean isAborted() {
+            return false;
+          }
+        });
+      peerClusterId = ZKClusterId.readClusterIdZNode(zkWatcher);
+    } catch (IOException | KeeperException e) {
+      throw new DoNotRetryIOException("Can't get peerClusterId for clusterKey=" + clusterKey, e);
+    }
```

Comment:
We want to close() the zkWatcher? Otherwise there is a connection leak?

Reply:
Good catch. Let me fix this.